oml/utils/misc.py

```diff
@@ -162,6 +163,40 @@ def pad_array_right(arr: np.ndarray, required_len: int, val: Union[float, int])
     return np.pad(arr, (0, required_len - len(arr)), mode="constant", constant_values=val)
 
 
+def visualise_text(text: str, color: TColor) -> np.ndarray:
+    im_size = 256
+    img = Image.new("RGB", (im_size, im_size), "white")
+    draw = ImageDraw.Draw(img)
+
+    font = ImageFont.load_default()
+
+    # Calculate the position for the text
+    margin = 10
+    max_width = im_size - 2 * margin
+    text_lines = []
+
+    # Split the text into lines that fit within the max_width
+    for line in text.split("\n"):
+        words = line.split()
+        current_line = []  # type: ignore
+        while words:
+            while words and draw.textlength(" ".join(current_line + [words[0]]), font=font) <= max_width:
```

Comment:
you enter an infinite loop if you can't fit your word into a line of pixels with the provided font

```python
plt.imshow(visualise_text("a" * 60, (150, 150, 150)))
plt.show()
```

I don't know how realistic such a situation is, but missing 1-2 spaces and getting a long single word is not impossible, especially for raw texts

Reply:
I just wanted to check how it looks and it freezes with long word

Reply:
you are right

I tried to cover all corner cases in the current function but gave up.

Finally, I reworked it using matplotlib and added a few tests on huge inputs.